File: src/main/java/com/sogeti/codingchallenge/day25/Day25Challenge.java
package com.sogeti.codingchallenge.day25;

import com.sogeti.codingchallenge.Challenge;

import java.util.List;

public abstract class Day25Challenge extends Challenge {

    Day25Challenge(String challengeTitle, String description) {
        super(challengeTitle, description, "20201209-input-day9.txt");
    }

    @Override
    protected String solve(List<String> inputList) {


        return getMessage(0);
    }




    abstract String getMessage(long value);
}

File: src/main/java/hzt/aoc/day25/Day25Challenge.kt
package hzt.aoc.day25

import hzt.aoc.Challenge

abstract class Day25Challenge internal constructor(challengeTitle: String, description: String) :
    Challenge(challengeTitle, description, "20201225-input-day25.txt") {
    override fun solve(inputList: List<String>): String {
        val cardPublicKey = inputList[0].toLong()
        val doorPublicKey = inputList[1].toLong()
        return getMessage(solveByInput(cardPublicKey, doorPublicKey))
    }

    protected abstract fun solveByInput(cardPublicKey: Long, doorPublicKey: Long): Long
    abstract fun getMessage(value: Long): String

    companion object {
        const val NUMBER_TO_DIVIDE_BY = 20201227
        const val INIT_SUBJECT_NUMBER = 7
    }
}
